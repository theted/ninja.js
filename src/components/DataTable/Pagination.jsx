import React from 'react';
import PropTypes from 'prop-types';

import Page from './Page';

const Pagination = ({ currentPageNumber, totalNumberOfPages, onChange }) => {
  const pages = Array.from(Array(totalNumberOfPages).keys()).map((pageNumber) => (
    <Page
      key={pageNumber}
      currentPageNumber={currentPageNumber}
      pageNumber={pageNumber}
      onChange={onChange}
    />
  ));

  if (pages.length <= 1) {
    return null;
  }
  return <ul className="pagination">{pages}</ul>;
};

Pagination.propTypes = {
  currentPageNumber: PropTypes.number.isRequired,
  totalNumberOfPages: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Pagination;
