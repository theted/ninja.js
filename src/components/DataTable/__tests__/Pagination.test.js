import { shallow } from 'enzyme';
import React from 'react';
import Pagination from '../Pagination';

describe('Pagination component', () => {
  it('renders without crashing', () => {
    shallow(<Pagination />);
  });

  it('matches snapshot', () => {
    const component = shallow(<Pagination />);
    expect(component).toMatchSnapshot();
  });
});
