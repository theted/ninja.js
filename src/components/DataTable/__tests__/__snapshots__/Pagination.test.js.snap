// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches snapshot 1`] = `ShallowWrapper {}`;
